droidlet/interpreter/robot/default_behaviors.py

```diff
@@ -36,18 +37,29 @@ def init_logger():
 
 init_logger()
 
+#TODO: Move these utils to a suitable place - as a class method in TripleNode
+def add_or_replace(agent, pred_text, obj_text):
+    memids, _ = agent.memory.basic_search(f'SELECT uuid FROM Triple WHERE pred_text={pred_text}')
+    assert len(memids) <= 1, f"more than 1 {len(memids)} returned"
+    if len(memids) > 0:
+        agent.memory.forget(memids[0])
+    agent.memory.add_triple(subj=agent.memory.self_memid, pred_text=pred_text, obj_text=obj_text)
+
+def get_unique_val_from_memory(agent, pred_text, typ):
+    
+    def get_default(typ):
+        if typ == int:
+            return 0
+        if typ == str:
+            return ''
+
+    t = agent.memory.get_triples(subj=agent.memory.self_memid, pred_text=pred_text)
+    assert len(t) <= 1, f"More than 1 ({len(t)}) triple for {pred_text}"
+    return typ(t[0][2]) if len(t) == 1 else get_default(typ)
```

Comment:
why `[0][2]`? add a comment on why the `2`

Reply:
`get_triples` returns a list of tuples of the form `(subject, predicate, object)`, hence the second index of the object. Ideally, I'd like a different interface for something like this which just returns the object without the caller having to index a tuple for it. Will do that as a follow-up. Adding a comment for now.